### PR TITLE
LRAC-10162  [C] Allow for users to pass in the param assetType into the attributes and options. & LRAC-10145 [C] Add error for incorrect data type for analytics.track event name

### PR DESCRIPTION
**Pull request by kienD**
Jira Issue: [LRAC-10162](https://issues.liferay.com/browse/LRAC-10162) | [LRAC-10145](https://issues.liferay.com/browse/LRAC-10145)

### CODE REVIEW

**Review by interaminense — APPROVED** [review]
LGTM

File: modules/apps/analytics/analytics-client-js/test/analytics.js [hunk 1]
```diff
@@ -290,6 +296,52 @@ describe('Analytics', () => {
 			]);
 		});
 
+		it('returns a type error if the eventId is not a string', () => {
+			const eventId = {test: 'test'};
+
+			console.error = jest.fn((val) => val);
+
+			Analytics.track(eventId);
```
Comment (gabrielpramos) [inline modules/apps/analytics/analytics-client-js/test/analytics.js:304]: This `Analytics.track` might need to have an await oparator, once `addItem` function is returning a promise after we fixed this bug
https://github.com/brianchandotcom/liferay-portal/pull/111121

Comment (gabrielpramos) [reply modules/apps/analytics/analytics-client-js/test/analytics.js:304]: So, this whole test case should be async

File: modules/apps/analytics/analytics-client-js/test/analytics.js [hunk 2]
```diff
@@ -290,6 +296,52 @@ describe('Analytics', () => {
...
+
+			expect(console.error).toHaveBeenCalledTimes(1);
+		});
+
+		it('uses the applicationId from options', () => {
+			const eventId = 'test';
+			const applicationId = 'Page';
+			const properties = {a: 1, b: 2, c: 3};
+
+			Analytics.track(eventId, properties, {applicationId});
```
Comment (gabrielpramos) [inline modules/apps/analytics/analytics-client-js/test/analytics.js:314]: Same as the one above

File: modules/apps/analytics/analytics-client-js/test/analytics.js [hunk 3]
```diff
@@ -290,6 +296,52 @@ describe('Analytics', () => {
...
+				}),
+			]);
+		});
+
+		it('uses the assetType from properties over the applicationId from options', () => {
+			const assetType = 'Blog';
+			const eventId = 'test';
+			const properties = {a: 1, assetType};
+
+			Analytics.track(eventId, properties, {applicationId: 'Page'});
```
Comment (gabrielpramos) [inline modules/apps/analytics/analytics-client-js/test/analytics.js:332]: Same here

**Review by gabrielpramos — COMMENTED** [review]
Good Job @kienD !
Just one quick question!

File: modules/apps/analytics/analytics-client-js/test/analytics.js (hunk 1 above)
Comment (kienD) [reply modules/apps/analytics/analytics-client-js/test/analytics.js:304]: Ok, I'll wait for your PR to be merged first.

**Comment by kienD** [comment]
@gabrielpramos updated!

**Comment by kienD** [comment]
ci:test

**Comment by kienD** [comment]
ci:test:relevant

**Comment by kienD** [comment]
ci:test:sf

File: modules/apps/analytics/analytics-client-js/src/utils/constants.js [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 // AC Version
 
-export const ANALYTICS_CLIENT_VERSION = '1.0.1';
```
Comment (gabrielpramos) [inline modules/apps/analytics/analytics-client-js/src/utils/constants.js:17]: FYI I didn't need to update the client version to the bug we fix

**Review by gabrielpramos — APPROVED** [review]
LGTM :fire:

**Comment by gabrielpramos** [comment]
ci:forward

File: modules/apps/analytics/analytics-client-js/src/utils/constants.js (hunk 4 above)
Comment (kienD) [reply modules/apps/analytics/analytics-client-js/src/utils/constants.js:17]: O I updated it because of the new release version.

**Pull request closed without merging**